pkg/testcore/suites/perf-suites.go

```diff
@@ -1087,14 +1087,25 @@ func (vis *VolumeIoSuite) Run(ctx context.Context, storageClass string, clients
 				}
 
 				if i != 0 {
-					writer := bytes.NewBufferString("")
-					if err := podClient.Exec(ctx, writerPod.Object, []string{"/bin/bash", "-c", "sha512sum -c " + sum}, writer, os.Stderr, false); err != nil {
-						return err
+					// Occasionally the sha512sum function returns empty even when the driver has correctly written data. This retry logic allows the sha512sum function to run again and correctly calculate the hash value.
```

Comment:
When we have duplicate code in more than 2 places, we should put those in a function and call the function. That will make sure we have shorter code, and more importantly, if any changes needed to be done in future, it will be done in one place and chance of missing in duplicate code is avoided by design.